File: src/main/intellibitz/sted/widgets/SplashWindow.java
package intellibitz.sted.widgets;

import intellibitz.sted.event.IStatusListener;
import intellibitz.sted.event.StatusEvent;

import javax.swing.*;
import java.awt.*;

public class SplashWindow
        extends JWindow
        implements IStatusListener {
    private final JProgressBar progress;

    public SplashWindow(Component component) {
        final Container contentPane = getContentPane();
        contentPane.setLayout(new BoxLayout(contentPane, BoxLayout.Y_AXIS));
        contentPane.add(component);
        progress = new JProgressBar(0, 100);
        progress.setStringPainted(true);
        contentPane.add(progress);
        pack();
    }

    public void setProgress(int percent) {
        progress.setValue(percent);
    }

    public void statusPosted(StatusEvent event) {
        setProgress(Integer.valueOf(event.getStatus()));
    }
}

File: src/main/intellibitz/sted/widgets/SplashWindow.kt
package sted.widgets

import sted.event.IStatusListener
import sted.event.StatusEvent
import java.awt.Component
import javax.swing.BoxLayout
import javax.swing.JProgressBar
import javax.swing.JWindow

class SplashWindow(component: Component?) : JWindow(), IStatusListener {
    private val progress: JProgressBar
    fun setProgress(percent: Int) {
        progress.value = percent
    }

    override fun statusPosted(event: StatusEvent) {
        event.status?.toInt()?.let { setProgress(it) }
    }

    init {
        val contentPane = contentPane
        contentPane.layout = BoxLayout(contentPane, BoxLayout.Y_AXIS)
        contentPane.add(component)
        progress = JProgressBar(0, 100)
        progress.isStringPainted = true
        contentPane.add(progress)
        pack()
    }
}